utils/sample_qc.py

```diff
@@ -364,82 +364,139 @@ def infer_sex(
+    :param float f_stat_XX_cutoff: Min f-stat for XX samples
+    :param flat f_stat_XY_cutoff: Max f-stat for XY samples
+    :param int lower_SD_cutoff: Number of standard deviations to use when determining upper limit of sex chromosome ploidy cutoffs
+    :param int upper_SD_cutoff: Permissive number of standard deviations to use when determining lower ploidy limit and aneuploidy cutoffs
+    :param int Y_lower_bound: Number of standard deviations to use as the lower bound for Y, specifically to correct for somatic loss of Y
     :rtype: StructExpression
     """
+    def get_ploidy_cutoffs(ht: hl.Table, sex: str, f_stat_cutoff: float, lower_SD_cutoff: int, upper_SD_cutoff: int, Y_lower_bound: int) -> dict:
+        """
+        Gets chromosome X and Y ploidy cutoffs for XY and XX samples
+
+        :param Table ht: Table with samples and their sex chromosome ploidies
+        :param str sex: Sex for which to check X/Y ploidies. Assumes input is either XX or XY.
+        :param float f_stat_cutoff: f-stat cutoff value. Assumes XX values will be above cutoff and XY values will be below cutoff.
+        :param int lower_SD_cutoff: Number of standard deviations to use when determining upper limit of sex chromosome ploidy cutoffs
+        :param int upper_SD_cutoff: Permissive number of standard deviations to use when determining lower ploidy limit and aneuploidy cutoffs
+        :param int Y_lower_bound: Number of standard deviations to use as the lower bound for Y, specifically to correct for somatic loss of Y
+        :return: Dictionary of ploidy cutoffs; key: sex chrom, values: [lower cut, upper cut, outlier (aneuploidy) cut]
+        :rtype: dict
+        """
+        if sex == "XX":
+            ht = ht.filter(ht.f_stat < f_stat_cutoff)
+        else:
+            ht = ht.filter(ht.f_stat > f_stat_cutoff)
+
+        # Get mean/stdev for chrX/Y ploidies
+        X_stats = ht.aggregate(hl.agg.stats(ht.chrX_ploidy))
+        Y_stats = ht.aggregate(hl.agg.stats(ht.chrY_ploidy))
+
+        # Set cutoffs -- keep cutoffs permissive to account for loss of sex chromosomes
+        cutoffs = {}
+        cutoffs["X"] = [
+                        X_stats.mean - (upper_SD_cutoff * X_stats.stdev), X_stats.mean + (lower_SD_cutoff * X_stats.stdev), 
+                        X_stats.mean + (upper_SD_cutoff * X_stats.stdev)
+                        ]
+        cutoffs["Y"] = [
+                        abs(min((Y_lower_bound * Y_stats.stdev), Y_stats.mean - (upper_SD_cutoff * Y_stats.stdev))), 
+                        Y_stats.mean + (lower_SD_cutoff * Y_stats.stdev), 
+                        Y_stats.mean + (upper_SD_cutoff * Y_stats.stdev)
+                        ]
+        return cutoffs
+   
+ 
+    xx_cutoffs = get_ploidy_cutoffs(ht, "XX", f_stat_XX_cutoff, lower_SD_cutoff, upper_SD_cutoff, Y_lower_bound)
+    xy_cutoffs = get_ploidy_cutoffs(ht, "XY", f_stat_XY_cutoff, lower_SD_cutoff, upper_SD_cutoff, Y_lower_bound)
```

Comment:
Add argue that here, it would make more sense  to get  the `X` and `Y` cutoffs for the different ploidies in this function rather than the cutoffs for males vs females. The arguments would be very much the same but then what you'd get would be an array where the index is the ploidy for that sex chrom and the value the (lower, upper) cutoffs.
So something like
```
x_cutoffs = get_ploidy_cutoffs(ht, "X", f_stat_XX_cutoff, f_stat_XY_cutoff, lower_SD_cutoff, upper_SD_cutoff)
```
which would return something like this for a perfect 20x cohort (so 10x depth per copy of any contig), with a cutoff set to an mean + SD value of 1,  2 and 3 for 1, 2 and 3 X copies resp. 
```
[(0, 0), (9, 11), (18, 22), (27, 33)]
```
Note that the first element could also be `NA` since I believe that no copy of X is a karyotype that's not compatible with life.

Similarly for Y, it would look like:
```
```
y_cutoffs = get_ploidy_cutoffs(ht, "Y", f_stat_XX_cutoff, f_stat_XY_cutoff, lower_SD_cutoff, upper_SD_cutoff, Y_lower_bound)
```
Which  would look like:
```
[(0, 1), (7, 11), ()]
```

I  think this is conceptually more pleasing than estimating cutoffs for both chroms based on male female and then have another layer of mapping the male/female to copy number cutoffs.

Reply:
I implemented something similar to this suggestion, where the cutoffs for X are:
```
[ (single X ploidy lower cutoff, single X ploidy upper cutoff), (double X ploidy lower cutoff, double X ploidy upper cutoff), triple X ploidy lower cutoff]
```
and the cutoffs for Y are:
```
[single Y ploidy lower cutoff, single Y ploidy upper cutoff, double Y ploidy lower cutoff]
```
This made a little more sense to me, but I'm happy to discuss and rework this